app/lib/actions/proxy_action.rb

```diff
@@ -146,22 +168,15 @@ def metadata=(thing)
       output[:metadata] = thing
     end
 
-    def timeout_set?
-      !metadata[:timeout].nil?
-    end
-
-    def set_timeout!
-      time = Time.zone.now + input[:connection_options][:timeout]
-      schedule_timeout(time)
-      metadata[:timeout] = time.to_s
-    end
-
     def default_connection_options
       # Fails if the plan is not finished within 60 seconds from the first task trigger attempt on the smart proxy
       # If the triggering fails, it retries 3 more times with 15 second delays
       { :retry_interval => Setting['foreman_tasks_proxy_action_retry_interval'] || 15,
-        :retry_count    => Setting['foreman_tasks_proxy_action_retry_count'] || 4,
-        :timeout        => Setting['foreman_tasks_proxy_action_start_timeout'] || 60 }
+        :retry_count    => Setting['foreman_tasks_proxy_action_retry_count'] || 4 }
+    end
+
+    def clean_remote_task(*_args)
+      remote_task.destroy!
```

Comment:
I don't know what exactly lead to this, but I got into situation, where this failed with 

```
Failed to run hook 'clean_remote_task' for action 'Actions::ProxyAction'
undefined method `destroy!' for nil:NilClass (NoMethodError)
 | /home/inecas/Projects/ws/foreman-rex/foreman-tasks/app/lib/actions/proxy_action.rb:179:in `clean_remote_task'
 | /home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/execution_plan/hooks.rb:68:in `block (2 levels) in run'
 | /home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/execution_plan/hooks.rb:66:in `each'
 | /home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/execution_plan/hooks.rb:66:in `block in run'
 | /home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/executors.rb:11:in `run_user_code'
 | /home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/execution_plan/hooks.rb:65:in `run'
 | /home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/execution_plan.rb:145:in `block in run_hooks'
 | /home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/execution_plan.rb:144:in `each'
 | /home/inecas/Projects/ws/foreman-rex/dynflow/lib/dynflow/execution_plan.rb:144:in `run_hooks'
```